metadata-service/factories/src/main/java/com/linkedin/gms/factory/entity/RetentionServiceFactory.java

```diff
@@ -23,19 +27,27 @@ public class RetentionServiceFactory {
   @Qualifier("entityService")
   private EntityService _entityService;
 
-  @Autowired
-  @Qualifier("ebeanServer")
-  private EbeanServer _server;
-
   @Value("${RETENTION_APPLICATION_BATCH_SIZE:1000}")
   private Integer _batchSize;
 
 
+  @Bean(name = "retentionService")
+  @DependsOn({"cassandraSession", "entityService"})
+  @ConditionalOnProperty(name = "ENTITY_SERVICE_IMPL", havingValue = "cassandra")
+  @Nonnull
+  protected RetentionService createCassandraInstance(CqlSession session) {
+    RetentionService retentionService = new CassandraRetentionService(_entityService, session, _batchSize);
```

Comment:
Huge improvement to the retention service, to decouple from Ebean. Awesome work.